src/Features/Core/Portable/CodeRefactorings/InvertIf/AbstractInvertIfCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,392 @@
+                    case BinaryOperatorKind.ConditionalAnd:
+                    case BinaryOperatorKind.ConditionalOr:
+                        negateOperands = true;
+                        break;
+                }
+
+                //Workaround for https://github.com/dotnet/roslyn/issues/23956
+                //Issue to remove this when above is merged
+                if (binaryOperation.OperatorKind == BinaryOperatorKind.Or && IsConditionalOr(operatorToken))
+                {
+                    negatedKind = BinaryOperatorKind.ConditionalAnd;
+                }
+                else if (binaryOperation.OperatorKind == BinaryOperatorKind.And && IsConditionalAnd(operatorToken))
+                {
+                    negatedKind = BinaryOperatorKind.ConditionalOr;
+                }
+
+                SyntaxNode newLeftOperand = null;
+                SyntaxNode newRightOperand = null;
+                if (negateOperands)
+                {
+                    newLeftOperand = Negate(leftOperand, generator, syntaxFacts, semanticModel, cancellationToken);
+                    newRightOperand = Negate(rightOperand, generator, syntaxFacts, semanticModel, cancellationToken);
+                }
+                else
+                {
+                    newLeftOperand = leftOperand;
+                    newRightOperand = rightOperand;
+                }
+
+                var newBinaryExpressionSyntax = NewBinaryOperation(binaryOperation, newLeftOperand, negatedKind, newRightOperand, generator, syntaxFacts, cancellationToken)
+                    .WithLeadingTrivia(leadingExpressionTrivia)
+                    .WithTrailingTrivia(trailingExpressionTrivia);
+
+                var newToken = syntaxFacts.GetOperatorTokenOfBinaryExpression(newBinaryExpressionSyntax);
+                var newTokenWithTrivia = newToken.WithLeadingTrivia(leadingTokenTrivia).WithTrailingTrivia(trailingTokenTrivia);
+                return newBinaryExpressionSyntax.ReplaceToken(newToken, newTokenWithTrivia);
+            }
+        }
+
+        private SyntaxNode NewBinaryOperation(
+            IBinaryOperation binaryOperation, 
+            SyntaxNode leftOperand, 
+            BinaryOperatorKind operationKind, 
+            SyntaxNode rightOperand, 
+            SyntaxGenerator generator, 
+            ISyntaxFactsService syntaxFacts, 
+            CancellationToken cancellationToken)
+        {
+            switch (operationKind)
+            {
+                case BinaryOperatorKind.Equals:
+                    if (binaryOperation.LeftOperand.Type?.IsValueType == true && binaryOperation.RightOperand.Type?.IsValueType == true)
+                    {
+                        return generator.ValueEqualsExpression(leftOperand, rightOperand);
+                    }
+                    else
+                    {
+                        return generator.ReferenceEqualsExpression(leftOperand, rightOperand);
+                    }
```

Comment:
nit: use ternary expression:   i.e. ```return ... ? ... : ...```

Reply:
you fixed up all the rest *except* this one :)

Reply:
doh!  fixed it now :)